indexer-core/src/main/java/org/apache/maven/index/updater/IndexDataReader.java

```diff
@@ -165,8 +165,8 @@ private IndexDataReadResult readIndexMT( IndexWriter w, IndexingContext context
 
         final Document theEnd = new Document();
 
-        ConcurrentMap<String, Boolean> rootGroups = new ConcurrentHashMap<>();
-        ConcurrentMap<String, Boolean> allGroups = new ConcurrentHashMap<>();
+        Set<String> rootGroups = ConcurrentHashMap.newKeySet();
+        Set<String> allGroups = ConcurrentHashMap.newKeySet();
```

Comment:
I tried to size them appropriately (central has about 70k groups, and resizing CHMs has a larger impact than normal maps), but since this made no performance difference I left the defaults as is.